general/isockstream.cpp

```diff
@@ -87,7 +87,7 @@ int isockstream::establish()
 #if defined(__APPLE__)
       if (bind(sfd, (const struct sockaddr *)rp->ai_addr, rp->ai_addrlen) < 0)
 #else
-      if (bind(sfd, rp->ai_addr, rp->ai_addrlen) < 0)
+      if (bind(sfd, rp->ai_addr, static_cast<int>(rp->ai_addrlen)) < 0)
```

Comment:
On Linux the last argument is not `int` but `socklen_t` (and that is defined as `unsigned int`, at least in some cases). MSVC does not define `socklen_t`, however we do that in the beginning of this file with `socklin_t` = `int`. Therefore, it will be best to cast to `socklen_t` here which will be the right type in all cases.